streams/src/main/java/org/apache/kafka/streams/state/internals/KeyValueByteProxyStore.java

```diff
@@ -0,0 +1,117 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.KeyValue;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+import java.util.List;
+
+import static org.apache.kafka.streams.state.TimestampedBytesStore.convertToTimestampedFormat;
+import static org.apache.kafka.streams.state.internals.ValueAndTimestampDeserializer.rawValue;
+
+public class KeyValueByteProxyStore implements KeyValueStore<Bytes, byte[]> {
+    final KeyValueStore<Bytes, byte[]> store;
+
+    KeyValueByteProxyStore(final KeyValueStore<Bytes, byte[]> store) {
+        this.store = store;
+    }
+
+    @Override
+    public void put(final Bytes key,
+                    final byte[] valueWithTimestamp) {
+        store.put(key, valueWithTimestamp == null ? null : rawValue(valueWithTimestamp));
+    }
+
+    @Override
+    public byte[] putIfAbsent(final Bytes key,
+                              final byte[] valueWithTimestamp) {
+        return convertToTimestampedFormat(store.putIfAbsent(
+            key,
+            valueWithTimestamp == null ? null : rawValue(valueWithTimestamp)));
+    }
+
+    @Override
+    public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
+        for (final KeyValue<Bytes, byte[]> entry : entries) {
+            final byte[] valueWithTimestamp = entry.value;
+            store.put(entry.key, valueWithTimestamp == null ? null : rawValue(valueWithTimestamp));
+        }
+    }
+
+    @Override
+    public byte[] delete(final Bytes key) {
+        return convertToTimestampedFormat(store.delete(key));
+    }
+
+    @Override
+    public String name() {
+        return store.name();
+    }
+
+    @Override
+    public void init(final ProcessorContext context,
+                     final StateStore root) {
+        store.init(context, root);
+    }
+
+    @Override
+    public void flush() {
+        store.flush();
+    }
+
+    @Override
+    public void close() {
+        store.close();
+    }
+
+    @Override
+    public boolean persistent() {
+        return true;
```

Comment:
Is this proxy only for RocksDB stores? Why not return store.persistent()?

Reply:
Yes, it's only for persistent stores. If a store is not persistent, there is no need to proxy it because we get the correct data from the changelog topic on restore, and now that the stored value bytes are in `timestamp+value` format.

I'll add a check to the constructor.

Reply:
Right, thanks for clarifying